master/buildbot/util/__init__.py

```diff
@@ -49,17 +49,34 @@ def key_func(item):
     return l
 
 
-def flatten(l, types=list):
-    if l and isinstance(l, types):
-        rv = []
-        for e in l:
-            if isinstance(e, types):
-                rv.extend(flatten(e))
-            else:
-                rv.append(e)
-        return rv
-    else:
-        return l
+def flattened_iterator(l, types=(list, tuple)):
```

Comment:
If this function is expected to be used only by `flatten` below, I'd suggest to mark it private `_flattened_iterator`; and in this case default value for `types`` does not seem to be necessary.